# This is a BitKeeper generated diff -Nru style patch.
#
# ChangeSet
#   2005/03/08 08:18:50-08:00 torvalds@ppc970.osdl.org
#   epoll: return proper error on overflow condition
#
#   Noted by Georgi Guninski.
#
# fs/eventpoll.c
#   2005/03/08 08:18:40-08:00 torvalds@ppc970.osdl.org +2 -1
#   epoll: return proper error on overflow condition
#
#   Noted by Georgi Guninski.
#
--- linux-2.6.10/fs/eventpoll.c.orig	2004-12-24 22:34:31.000000000 +0100
+++ linux-2.6.10/fs/eventpoll.c	2005-03-09 23:53:17.995842016 +0100
@@ -619,6 +619,7 @@
 	return error;
 }
 
+#define MAX_EVENTS (INT_MAX / sizeof(struct epoll_event))
 
 /*
  * Implement the event wait interface for the eventpoll file. It is the kernel
@@ -635,7 +636,7 @@
 		     current, epfd, events, maxevents, timeout));
 
 	/* The maximum number of event must be greater than zero */
-	if (maxevents <= 0)
+	if (maxevents <= 0 || maxevents > MAX_EVENTS)
 		return -EINVAL;
 
 	/* Verify that the area passed by the user is writeable */
